### ISSUE TITLE
test suite should show grand total of total/passed/skipped/failed ; currently it's misleading

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#8639]
currently, if I run test suite on OSX via `./koch tests all` (and log to a file), I get:

`grep failed: /Users/timothee/temp//logs//run_with_log.D20180814T001204.log`

FAILURE! total: 33 passed: 32 skipped: 0 failed: 1
FAILURE! total: 24 passed: 22 skipped: 0 failed: 2
FAILURE! total: 21 passed: 18 skipped: 2 failed: 1
FAILURE! total: 5 passed: 3 skipped: 0 failed: 2
FAILURE! total: 4 passed: 2 skipped: 0 failed: 2
FAILURE! total: 1 passed: 0 skipped: 0 failed: 1
FAILURE! total: 105 passed: 85 skipped: 0 failed: 20

this file ends with "FAILURE! total: 105 passed: 85 skipped: 0 failed: 20" ; and one may be tempted to think it's a grand total of all tests; but it's not:
the numbers don't add up: 33+24+21+5+4+1 = 88 instead of 105, etc.

## proposal
* tests should show a grand total of all tests after all tests are run, eg:
* instead of `FAILURE! total: 24 passed: 22 skipped: 0` only upon failure, show the more informative:
`TOTAL(compilerapi):FAILURE total: 24 passed: 22 skipped: 0`
or, if success:`TOTAL(compilerapi):SUCCESS total: 24 passed: 24 skipped: 0`
which is still as easily grepable for failures, but more informative as to what failed upon failure.

* `failed:` shall be printed => PR https://github.com/nim-lang/Nim/pull/8631

```
...
TOTAL(longgc):FAILURE total: 33 passed: 32 skipped: 0 failed: 1
TOTAL(compilerapi):FAILURE total: 1 passed: 0 skipped: 0 failed: 1
...
GRAND TOTAL:FAILURE total: 193 passed: 163 skipped: 0 failed: 29
```

**Comment by timotheecour** [comment]
yes, it is another category of tests; it's just the last one shown and the 1st time i saw it I assumed (wrongly) it was a summary of all tests being run (I'm probably not the only one who got confused about that).
in any case, the other points hold IMO

**Comment by Araq** [comment]
Yeah it's not a summary. A summary is somewhat hard to implement as the tester runs the categories in parallel.
